kbl2vec/src/main/java/com/foursoft/harness/kbl2vec/transform/geometry/geo_2d/GeometrySegment2DTransformer.java

```diff
@@ -0,0 +1,73 @@
+/*-
+ * ========================LICENSE_START=================================
+ * KBL to VEC Converter
+ * %%
+ * Copyright (C) 2025 4Soft GmbH
+ * %%
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ * =========================LICENSE_END==================================
+ */
+package com.foursoft.harness.kbl2vec.transform.geometry.geo_2d;
+
+import com.foursoft.harness.kbl.v25.KblSegment;
+import com.foursoft.harness.kbl2vec.convert.DoublesToCartesianVector2DConverter;
+import com.foursoft.harness.kbl2vec.core.Query;
+import com.foursoft.harness.kbl2vec.core.TransformationContext;
+import com.foursoft.harness.kbl2vec.core.TransformationResult;
+import com.foursoft.harness.kbl2vec.core.Transformer;
+import com.foursoft.harness.kbl2vec.transform.geometry.GeometryDimensionDetector;
+import com.foursoft.harness.vec.v2x.VecGeometryNode2D;
+import com.foursoft.harness.vec.v2x.VecGeometrySegment2D;
+import com.foursoft.harness.vec.v2x.VecTopologySegment;
+
+public class GeometrySegment2DTransformer implements Transformer<KblSegment, VecGeometrySegment2D> {
+
+    @Override
+    public TransformationResult<VecGeometrySegment2D> transform(final TransformationContext context,
+                                                                final KblSegment source) {
+        final VecGeometrySegment2D destination = new VecGeometrySegment2D();
+        destination.setIdentification(source.getId());
+
+        if (!GeometryDimensionDetector.hasDimensions(source.getStartVectors(), GeometryDimensionDetector.GEO_2D)) {
+            context.getLogger().warn(
+                    "Failed to transform start vectors of KblSegment (ID: {}). Expected 2 coordinates for 2D " +
```

Comment:
Failed is not correct here. You are still doing the transformation but the data might be inconsistent.